File: security-management-frontend/src/services/FicheDePaieService.js
// src/services/FicheDePaieService.js
import api from "./api"; // ou import axios from "axios";

const PATH = "/fiches-paie"; // Corrigé pour éviter la duplication de /api/

const FicheDePaieService = {
    getAll: ()             => api.get(PATH),
    getById: id            => api.get(`${PATH}/${id}`),
    create: dto            => api.post(PATH, dto),
    update: (id, dto)      => api.put(`${PATH}/${id}`, dto),
    remove: id             => api.delete(`${PATH}/${id}`)
};

export default FicheDePaieService;

File: SecurityManagementApp/src/main/java/com/boulevardsecurity/securitymanagementapp/service/FicheDePaieService.java
// src/main/java/com/boulevardsecurity/securitymanagementapp/service/FicheDePaieService.java
package com.boulevardsecurity.securitymanagementapp.service;

import com.boulevardsecurity.securitymanagementapp.dto.*;
import java.util.List;
import java.util.Optional;

public interface FicheDePaieService {
    FicheDePaieDto create(FicheDePaieCreationDto dto);
    Optional<FicheDePaieDto> getById(Long id);
    List<FicheDePaieDto> getAll();
    FicheDePaieDto update(Long id, FicheDePaieCreationDto dto);
    void delete(Long id);
}
